src/components/ContactSalesForm.tsx

```diff
@@ -0,0 +1,534 @@
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import * as z from "zod";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { Textarea } from "@/components/ui/textarea";
+import { Checkbox } from "@/components/ui/checkbox";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Loader2, CheckCircle, AlertCircle } from "lucide-react";
+
+// Disposable email domains for basic spam protection
+const disposableEmailDomains = [
+  "10minutemail.com",
+  "guerrillamail.com",
+  "mailinator.com",
+  "temp-mail.org",
+  "throwaway.email",
+];
+
+const contactSalesSchema = z.object({
+  companyName: z
+    .string()
+    .min(2, "Company name must be at least 2 characters")
+    .max(100, "Company name must be less than 100 characters"),
+  workEmail: z
+    .string()
+    .email("Please enter a valid email address")
+    .refine((email) => {
+      const domain = email.split("@")[1];
+      return !disposableEmailDomains.includes(domain?.toLowerCase());
+    }, "Please use a work email address"),
+  teamSize: z
+    .number({ required_error: "Team size is required" })
+    .int()
+    .min(1, "Team size must be at least 1")
+    .max(10000, "Team size must be less than 10,000"),
+  countryRegion: z.string().min(1, "Please select your country/region"),
+  hasConsent: z
+    .boolean()
+    .refine((val) => val === true, "You must agree to be contacted"),
+  phone: z
+    .string()
+    .optional()
+    .refine(
+      (val) => !val || val.length >= 10,
+      "Phone number must be at least 10 digits"
+    ),
+  useCase: z
+    .string()
+    .max(500, "Use case must be less than 500 characters")
+    .optional(),
+  preferredContactMethod: z.string().optional(),
+  preferredContactTime: z.string().optional(),
+  message: z
+    .string()
+    .max(1000, "Message must be less than 1000 characters")
+    .optional(),
+});
+
+type ContactSalesForm = z.infer<typeof contactSalesSchema>;
+
+const countries = [
+  "United States",
+  "Canada",
+  "United Kingdom",
+  "Germany",
+  "France",
+  "Australia",
+  "Japan",
+  "Singapore",
+  "Netherlands",
+  "Sweden",
+  "Norway",
+  "Denmark",
+  "Switzerland",
+  "Austria",
+  "Belgium",
+  "Finland",
+  "Ireland",
+  "New Zealand",
+  "South Korea",
+  "Other",
+];
```

Comment:
Any particular reason you are limiting to these countries only? Also, can you switch to a package that list out countries or move this to a constant file